test/test_dbc.cpp

```diff
@@ -49,4 +49,80 @@ TEST_CASE("Testing dbc file loading", "[fileio]") {
 		REQUIRE(parser->get_messages().front().signals == msg.signals);
 	}
 
-}
\ No newline at end of file
+}
+
+/*!
+ * \brief TEST_CASE
+ * Test negative values in offset, min, max
+ */
+TEST_CASE("Testing negative values") {
+  const auto* filename = std::tmpnam(NULL);
+
+  auto* file = std::fopen(filename, "w");
+  CHECK(file);
+
+  std::fputs(PRIMITIVE_DBC.c_str(), file);
+  std::fputs(R"(BO_ 234 MSG1: 8 Vector__XXX
+ SG_ Sig1 : 55|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig2 : 39|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig3 : 23|16@0- (10,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig4 : 7|16@0- (1,-10) [0|32767] "" Vector__XXX)", file);
+  std::fclose(file);
+
+  auto parser = libdbc::DbcParser();
+  parser.parse_file(std::string(filename));
+
+  REQUIRE(parser.get_messages().size() == 1);
+  REQUIRE(parser.get_messages().at(0).signals.size() == 4);
+  {
+    const auto signal = parser.get_messages().at(0).signals.at(0);
+    REQUIRE(signal.factor == 0.1);
+    REQUIRE(signal.offset == 0);
+    REQUIRE(signal.min == -3276.8);
+    REQUIRE(signal.max == -3276.7);
+  }
+  {
+    const auto signal = parser.get_messages().at(0).signals.at(1);
+    REQUIRE(signal.factor == 0.1);
+    REQUIRE(signal.offset == 0);
+    REQUIRE(signal.min == -3276.8);
+    REQUIRE(signal.max == -3276.7);
+  }
+  {
+    const auto signal = parser.get_messages().at(0).signals.at(2);
+    REQUIRE(signal.factor == 10);
+    REQUIRE(signal.offset == 0);
+    REQUIRE(signal.min == -3276.8);
+    REQUIRE(signal.max == -3276.7);
+  }
+  {
+    const auto signal = parser.get_messages().at(0).signals.at(3);
+    REQUIRE(signal.factor == 1);
+    REQUIRE(signal.offset == -10);
+    REQUIRE(signal.min == 0);
+    REQUIRE(signal.max == 32767);
+  }
+}
+
+
+TEST_CASE("Special characters in unit") {
+    const auto* filename = std::tmpnam(NULL);
+
+    auto* file = std::fopen(filename, "w");
+    CHECK(file);
+
+    std::fputs(PRIMITIVE_DBC.c_str(), file);
+    std::fputs(R"(BO_ 234 MSG1: 8 Vector__XXX
+ SG_ Speed : 0|8@1+ (1,0) [0|204] "Km/h"  DEVICE1,DEVICE2,DEVICE3)", file);
+    std::fclose(file);
+
+    auto parser = libdbc::DbcParser();
+    parser.parse_file(std::string(filename));
+
+    REQUIRE(parser.get_messages().size() == 1);
+    REQUIRE(parser.get_messages().at(0).signals.size() == 1);
+    {

```

Comment:
same suggestion here with the `SECTION` tag.

Reply:
changed in 6b4061c8759789bdc5138227d4f5fa564ef9304e